Remove dependency to devtools

thomashohn opened the pull request:
## Motivation/Description of the PR
- More info why we remove the dep to devtools of wdio helper: https://webdriver.io/blog/2024/08/15/webdriverio-v9-release/
- Resolves #4560.

Applicable helpers:

- [ ] Playwright
- [ ] Puppeteer
- [x] WebDriver
- [ ] REST
- [ ] FileHelper
- [ ] Appium
- [ ] TestCafe

Applicable plugins:

- [ ] allure
- [ ] autoDelay
- [ ] autoLogin
- [ ] customLocator
- [ ] pauseOnFail
- [ ] coverage
- [ ] retryFailedStep
- [ ] screenshotOnFail
- [ ] selenoid
- [ ] stepByStepReport
- [ ] stepTimeout
- [ ] wdio
- [ ] subtitles

## Type of change

- [ ] :fire: Breaking changes
- [ ] :rocket: New functionality
- [ ] :bug: Bug fix
- [x] 🧹 Chore
- [ ] :clipboard: Documentation changes/updates
- [ ] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [ ] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

thomashohn committed:
Removed some tests

thomashohn committed:
Removed reference

thomashohn committed:
Remove options

thomashohn committed:
Cleanup docs

thomashohn commented:
@kobenguyent - maybe you have an idea of the failing tests?
The idea was to have the tests running and then switch driver version

kobenguyent commented:
> @kobenguyent - maybe you have an idea of the failing tests? The idea was to have the tests running and then switch driver version

hi @thomashohn I reckoned you removed the devtools protocol but still calling this https://github.com/codeceptjs/CodeceptJS/pull/4563/files#diff-06a1e9f6d0b95cc16170ad023c4131c7d35c145e9c6375e492ab48a8b2fa10e4R646

I'd think that causes the issue. Also please check the impact of this removal as this may cause many issues related to this change.

thomashohn commented:
I guess I was uncertain what should happen to methods using
startRecordingTraffic and stopRecordingTraffic which use this.page

kobenguyent commented:
> I guess I was uncertain what should happen to methods using
> 
> startRecordingTraffic and stopRecordingTraffic which use this.page

If I'm not wrong, Those methods use the devtools protocol as well. 

Initially, you could refer to this PR which introduces devtools https://github.com/codeceptjs/CodeceptJS/pull/4105

thomashohn committed:
Fix tests

thomashohn committed:
Cleanup

thomashohn committed:
Fix types

thomashohn committed:
Cleanup and package.json update

thomashohn committed:
Back to previous versions

thomashohn committed:
Removed dependency to devtools

thomashohn committed:
Update webdriverio to latest 8.x version

thomashohn commented:
@kobenguyent - so Tests are green, removed the requirement to devtools and updated to latest webdriverio on v8 - there might be some documentation that needs adoptions?

kobenguyent commented:
> @kobenguyent - so Tests are green, removed the requirement to devtools and updated to latest webdriverio on v8 - there might be some documentation that needs adoptions?

Oh great! Thanks for your hard work! regarding docs, just edit the jsdocs, then run `npm run docs`, it would update the .md files for you.

thomashohn commented:
Yes - I was more unsure what to update exactly ;-) And the command seems to update a lot more than I changed ,,,

kobenguyent commented:
https://github.com/codeceptjs/CodeceptJS/blob/3.x/package.json#L93 perhaps we need to remove this as well.

thomashohn commented:
Should be gone: https://github.com/codeceptjs/CodeceptJS/pull/4563/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

kobenguyent commented:
![Screenshot 2024-11-12 at 22 55 02](https://github.com/user-attachments/assets/37aad9e6-176a-4da3-b0b7-e53dcebc5176)
just noticed some places.

thomashohn committed:
Cleanups and regenerate docs

kobenguyent commented on docs/changelog.md:
I think changelog could be left untouched.

thomashohn replied:
Hrm for some weird reason my revert attempt didn't work..

thomashohn replied:
Shouldn't it be git checkout 63b1d505 -- docs/changelog.md  ?

kobenguyent replied:
yeah, looks like that commit https://github.com/codeceptjs/CodeceptJS/commit/641e6746737181c6c045a72798a39004e315f2dc is correct.

thomashohn committed:
Leave changelog un-touched

thomashohn committed:
Second attempt

thomashohn committed:
Use correct hash

thomashohn committed:
Correct revert
